UPDATED THE UI: Live price for the homepage chart

Dprof-in-tech opened the pull request:
This pr resolves #49 

- [x] please use standard fetch, do not add axios or another library on top.
- [x] fetch the OP token price, later in the future we will need to add this as a config option but it's out of scope at the moment.
- [x] please stay within the scope of this issue which is very limited, in particular do not add loading indicators, auto refresh or additional controls to display another range as this is not needed at the moment.
- [x] use the shadcn/ui charts component
- [x] made sure there are no build errors or anything

Media files to show the result of this pr

![image](https://github.com/user-attachments/assets/872b4a72-4fe8-4ea6-afe2-83eab34d7b77)

Dprof-in-tech committed:
updated the ui live chart price for the optimism token

saimeunt commented on src/components/pages/home/transaction-history.tsx:
```diff
@@ -1,71 +1,122 @@
-  return {
-    name: format(date, "MMM d"),
-    date: format(date, "yyyy-MM-dd"),
-    transactionsCount: Math.floor(Math.random() * 200) + 400,
```
This placeholder datapoint is still needed, please reintroduce it.

saimeunt commented on src/components/pages/home/transaction-history.tsx:
```diff
@@ -1,71 +1,122 @@
+type DataPoint = {
+  name: string;
+  date: string;
+  price: number;
```
Missing the `transactionsCount` here.

Dprof-in-tech replied:
reintroduced chief

saimeunt commented on src/components/pages/home/transaction-history.tsx:
```diff
@@ -1,71 +1,122 @@
+  price: number;
+};
+
+const CustomTooltip = ({ active, payload }: any) => {
```
Why removing typing here? We don't want any, please reintroduce `TooltipProps<number, string>`.

Dprof-in-tech replied:
fixed

saimeunt commented on src/components/pages/home/transaction-history.tsx:
```diff
@@ -1,71 +1,122 @@
+        <p className="text-xs">
           {format(parseISO(payload[0].payload.date), "EEEE, MMMM d, yyyy")}
         </p>
-        <p>Transactions: {payload[0].value}k</p>
```
We do want the transactions count displayed in the tooltip.

saimeunt commented on src/components/pages/home/transaction-history.tsx:
```diff
@@ -1,71 +1,122 @@
+const TransactionHistory = () => {
+  const [data, setData] = useState<DataPoint[]>([]);
+
+  useEffect(() => {
```
Please avoid client-side data fetching as it's unnecessary here and can be refactored in using React Server Components.
Build the `DataPoint` array in the top level homepage component in `components/pages/home/index.tsx` and pass it as a prop in the `TransactionHistory` component.

saimeunt reviewed (changes requested):
@Dprof-in-tech this is heading in the right direction, nice work, please fix the small problems pointed out (reintroduce transactionsCount and use server-side data fetching) and we can merge this!

Dprof-in-tech committed:
used server side fetchign as in the code review

Dprof-in-tech committed:
Merge branch 'main' into main

Dprof-in-tech commented:
i've fixed the code as required in your review. here's a media file to show

![image](https://github.com/user-attachments/assets/244fcfba-bfd9-4e39-bbaf-83a93a1526ea)

saimeunt commented on src/components/pages/home/transaction-history.tsx:
```diff
@@ -28,54 +29,27 @@ type DataPoint = {
-    fetchData();
-  }, []);
+const TransactionHistory = ({ history }: { history: DataPoint[] }) => {
+  const [data, setData] = useState<DataPoint[]>(history);
```
I think this is no longer necessary and you can use the history prop directly.

Dprof-in-tech replied:
done chief

saimeunt reviewed (changes requested):
@Dprof-in-tech nice work on the requested changes, one final change requested and we can merge.

Dprof-in-tech committed:
using history directly

Dprof-in-tech committed:
Merge branch 'main' of https://github.com/Dprof-in-tech/op-scan
merge branch

Dprof-in-tech committed:
now using history prop directly

Dprof-in-tech committed:
Merge branch 'main' into main

saimeunt reviewed (changes requested):
@Dprof-in-tech You made the requested changes but your merge commit is wrong, we moved `fetchHomePageData` to a new function named `fetchHomeData` in `components/pages/home/fetch-home-data.ts`.

You must use this function in `components/pages/home/index.tsx` as `fetchHomePageData` has been removed from the codebase.
What you need to do instead is move your `fetchTransactionHistory` function to `components/pages/home/fetch-home-data.ts` and use it inside `fetchHomeData`.

Your commit should not introduce any changes to `lib/fetch-data.ts` as you currently have introduced regressions by using your outdated version of this file instead of the main branch version.

Dprof-in-tech commented:
Okay

Could you give me a few minutes to effect this change real quick so you
could take a look??

On Sun, Aug 4, 2024 at 12:02 AM saimeunt ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> @Dprof-in-tech <https://github.com/Dprof-in-tech> You made the requested
> changes but your merge commit is wrong, we moved fetchHomePageData to a
> new function named fetchHomeData in
> components/pages/home/fetch-home-data.ts.
>
> You must use this function in components/pages/home/index.tsx as
> fetchHomePageData has been removed from the codebase.
> What you need to do instead is move your fetchTransactionHistory function
> to components/pages/home/fetch-home-data.ts and use it inside
> fetchHomeData.
>
> Your commit should not introduce any changes to lib/fetch-data.ts as you
> currently have introduced regressions by using your outdated version of
> this file instead of the main branch version.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/walnuthq/op-scan/pull/50#pullrequestreview-2217240839>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3W3TPM236ATMG2Y3HPKJATZPVOSFAVCNFSM6AAAAABLYMVEF6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEMJXGI2DAOBTHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Dprof-in-tech committed:
refactored the code to fix the mistakes from the previous commit

Dprof-in-tech commented:
Hey chief
I just fixed the issue

Please review

Thank you

On Sun, Aug 4, 2024 at 12:34 AM Isaac Onyemaechi ***@***.***>
wrote:

> Okay
>
> Could you give me a few minutes to effect this change real quick so you
> could take a look??
>
> On Sun, Aug 4, 2024 at 12:02 AM saimeunt ***@***.***> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> @Dprof-in-tech <https://github.com/Dprof-in-tech> You made the requested
>> changes but your merge commit is wrong, we moved fetchHomePageData to a
>> new function named fetchHomeData in
>> components/pages/home/fetch-home-data.ts.
>>
>> You must use this function in components/pages/home/index.tsx as
>> fetchHomePageData has been removed from the codebase.
>> What you need to do instead is move your fetchTransactionHistory
>> function to components/pages/home/fetch-home-data.ts and use it inside
>> fetchHomeData.
>>
>> Your commit should not introduce any changes to lib/fetch-data.ts as you
>> currently have introduced regressions by using your outdated version of
>> this file instead of the main branch version.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/walnuthq/op-scan/pull/50#pullrequestreview-2217240839>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/A3W3TPM236ATMG2Y3HPKJATZPVOSFAVCNFSM6AAAAABLYMVEF6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDEMJXGI2DAOBTHE>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
